src/domain_decomposition.hpp

```diff
@@ -13,97 +16,91 @@
 namespace nest {
 namespace mc {
 
-// Meta data used to guide the domain_decomposition in distributing
-// and grouping cells.
-struct group_rules {
-    cell_size_type target_group_size;
-    backend_policy policy;
+inline bool has_gpu_backend(cell_kind k) {
```

Comment:
Here we have a helper that is used by the load balancing code to mark a cell_group for creation on either the host or device, if available.

The earlier `cell_kind::` code used the result of this decision, and generated the concrete instantion of the cell_group for the chosen back end.

This is the load balancing code that would make more nuanced decisions about how to partition work over cpu and gpu according to performance models. However for now, we simply place all work that can run on the gpu on the gpu if a gpu is available.